classes/class3/README.md

```diff
@@ -1 +1,364 @@
+Go ahead and copy that code into a file `main.handlebars` and put it in a new folder `layouts` within the `views` directory of your application.
+
+Now for the template itself. Starting off simple:
+```html
+<h1>Welcome Home!</h1>
+```
+Save that in a file called `home.handlebars` and put in the `views` directory as a sibling to the new `layouts` directory you just made. Your folder structure should look like this:
+```
+.
+├── app.js
+└── views
+    ├── home.handlebars
+    └── layouts
+        └── main.handlebars
+
+2 directories, 3 files
+```
+All of this organization is completely customizable, but we'll leave it as the defaults for now.
+
+Now let's hook it all up!
+
+Add the line `var exphbs  = require('express-handlebars');` to the top of your `app.js` file. Then add these two lins after the `var app = express();` line:
+```javascript
+app.engine('handlebars', exphbs({defaultLayout: 'main'}));
+app.set('view engine', 'handlebars');
+```
+Note that you're setting the default layout for the application to the `main.handlebars` file you just created. You can change the layout for a single template at render time with a simple option.
+
+Lastly, within the `index.js` routing file, replace:
+```javascript
+res.send("Welcome home!");
+```
+with
+```javascript
+res.render('home');
+```
+Restart your server, visit http://localhost:3000 and behold your template!
+
+At this point you are basically just writing plain HTML, let's fix that!
+Start by adding the following to your `home.handlebars` template:
+```html
+<ul>
+{{#each classes}}
+  <li>{{this}}</li>
+{{/each}}
+</ul>
+```
+Then change the contents of your `index.js` route again. This time to this:
+```javascript
+res.render("home", {"classes": [
+  "Olin.js",
+  "other class 1",
+  "other class 2",
+  "other class 3"]
+});
+```
+The second parameter that we are passing to the `render` function is the *context* for the template. That means that the word `classes` in the handlebars file is looking for an array named `classes` and it will create a new list item for *each* of the elements in the array. There's plenty more power in templates and we'll see them shortly.
+
+## MongoDB
+
+Node has the ability to store some information as variables. But all of the variables are stored in the process memory. If you restart the server (or the server crashes) the data will be wiped away. If you are storing information from users or information based on user input, we need a more permanent solution. That's very persistent storage comes in. A file, like `data.txt`, is a form of persistent storage. It can be written to, read from, and maintains information regardless of whether your server is running or has long since stopped. But files aren't useful for much more than storing log information because of their organization. When we want to retrieve data from persistent storage quickly, we need a solution like a database!
```

Comment:
very persistent -> where persistent
organization. -> organization and slow hard disk read/write speeds.